### PR TITLE
Upgrade to jnr-unixsocket 0.18

### PR DESCRIPTION
**Pull request by davidxia**
Use upstream UnixSocket instead of our own ApacheUnixSocket wrapper.

### CODE REVIEW

**Comment by caipre** [comment]
I guess the comment is no longer valid re: delaying the socket options?

https://github.com/spotify/docker-client/pull/918/files#diff-a479355e73762bc07b466a285d105088L41

I'm fine with this given that tests pass.

**Comment by davidxia** [comment]
@caipre Yea, I think `jnr.unixsocket.UnixSocket` does the right thing compared to `jnr.unixsocket.UnixSocketChannel`, but I haven't dug into it deeply. I just [noticed this as well](https://github.com/spotify/docker-client/pull/103/files#r151032803).

File: src/test/java/com/spotify/docker/client/UnixConnectionSocketFactoryTest.java [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+
+public class UnixConnectionSocketFactoryTest {
+
+  @Rule
+  public ExpectedException exception = ExpectedException.none();
+
+  private static UnixConnectionSocketFactory sut;
```
Comment (mattnworb) [inline src/test/java/com/spotify/docker/client/UnixConnectionSocketFactoryTest.java]: I don't think this needs to be `static` when it is reused in all `@Test`s

Comment (davidxia) [reply src/test/java/com/spotify/docker/client/UnixConnectionSocketFactoryTest.java]: ah yes, changed

**Pull request merged**